rpc/backend/tx_pool.go

```diff
@@ -1,43 +1,239 @@
+	if evmMempool == nil {
+		return inspect, nil
+	}
+
+	// Get pending (runnable) and queued (blocked) transactions from the mempool
+	pending, queued := evmMempool.GetTxPool().Content()
+
+	// Helper function to format transaction for inspection
+	format := func(tx *ethtypes.Transaction) string {
+		if to := tx.To(); to != nil {
+			return fmt.Sprintf("%s: %v wei + %v gas × %v wei",
+				tx.To().Hex(), tx.Value(), tx.Gas(), tx.GasPrice())
+		}
+		return fmt.Sprintf("contract creation: %v wei + %v gas × %v wei",
+			tx.Value(), tx.Gas(), tx.GasPrice())
+	}
+
+	// Flatten the pending transactions
+	for account, txs := range pending {
+		dump := make(map[string]string)
+		for _, tx := range txs {
+			dump[fmt.Sprintf("%d", tx.Nonce())] = format(tx)
+		}
+		inspect[StatusPending][account.Hex()] = dump
+	}
+
+	// Flatten the queued transactions
+	for account, txs := range queued {
+		dump := make(map[string]string)
+		for _, tx := range txs {
+			dump[fmt.Sprintf("%d", tx.Nonce())] = format(tx)
+		}
+		inspect[StatusQueued][account.Hex()] = dump
+	}
+
 	return inspect, nil
 }
 
 // Status returns the number of pending and queued transaction in the pool.
 func (b *Backend) Status() (map[string]hexutil.Uint, error) {
+	// Get the global mempool instance
+	evmMempool := mempool.GetGlobalEVMMempool()
+	if evmMempool == nil {
+		return map[string]hexutil.Uint{
+			StatusPending: hexutil.Uint(0),
+			StatusQueued:  hexutil.Uint(0),
+		}, nil
+	}
+
+	pending, queued := evmMempool.GetTxPool().Stats()
 	return map[string]hexutil.Uint{
-		"pending": hexutil.Uint(0),
-		"queued":  hexutil.Uint(0),
+		StatusPending: hexutil.Uint(pending), // #nosec G115 -- overflow not a concern for tx counts
+		StatusQueued:  hexutil.Uint(queued),  // #nosec G115 -- overflow not a concern for tx counts
 	}, nil
 }
+
+// convertToRPCTransaction converts an Ethereum transaction to RPC format for mempool display
+func (b *Backend) convertToRPCTransaction(tx *ethtypes.Transaction, from common.Address) (*types.RPCTransaction, error) {
+	curHeader, err := b.CurrentHeader()
+	if err != nil {
+		return nil, err
+	}
+	chainConfig := b.ChainConfig()
+
+	// Calculate base fee for pending transactions
+	var baseFee *big.Int
+	if curHeader != nil && curHeader.BaseFee != nil {
+		baseFee = curHeader.BaseFee
+	}
+
+	// Create RPC transaction directly
+	v, r, s := tx.RawSignatureValues()
+	rpcTx := &types.RPCTransaction{
+		Type:     hexutil.Uint64(tx.Type()),
+		From:     from,
+		Gas:      hexutil.Uint64(tx.Gas()),
+		GasPrice: (*hexutil.Big)(tx.GasPrice()),
+		Hash:     tx.Hash(),
+		Input:    hexutil.Bytes(tx.Data()),
+		Nonce:    hexutil.Uint64(tx.Nonce()),
+		To:       tx.To(),
+		Value:    (*hexutil.Big)(tx.Value()),
+		V:        (*hexutil.Big)(v),
+		R:        (*hexutil.Big)(r),
+		S:        (*hexutil.Big)(s),
+		ChainID:  (*hexutil.Big)(chainConfig.ChainID),
+	}
+
+	// Handle transaction type specific fields
+	switch tx.Type() {
+	case ethtypes.AccessListTxType:
```

Comment:
do we want a default case for invalid types?

Reply:
Not here, there are other valid types (like Legacy) that are valid and don't need to be changed.

Reply:
There should also never be invalid tx in the txPool as they would be rejected on attempted addition